tasks/src/reinforcement-learning/about.md

```diff
@@ -1,10 +1,76 @@
+
+Some of the autonomous driving tasks where reinforcement learning could be applied include trajectory optimization, motion planning, dynamic path finding, controller optimization, and scenario-based learning policies for highways. Reinforcement Learning agents are trained in dynamic environments to optimize trajectories. The agents are capable of accomplishing tasks such as motion planning, route changing, decision and position of parking and speed control.
+
+
+### NLP (Natural Language processing)
+
+NLP is a branch of machine learning or artificial intelligence concerned with allowing computers to understand the text and spoken words the same way as a human does.  
+
+Language understanding uses Reinforcement Learning because of its inherent nature of decision making it learns from the mistakes. The agent tries to understand the state of the sentence and tries to form an action set maximizing the value it would add.
+
+Reinforcement Learning is used in various NLP tasks such as text summarization, question answering, translation, dialogue generation, or machine translation.
+
+Reinforcement Learning agents can be trained to understand a few sentences of the document and use it to answer the corresponding questions.
+
+Robots in industries or healthcare working towards reducing manual intervention use reinforcement learning to map natural language instructions to sequences of executable actions. During training, the learner repeatedly constructs action sequences, executes those actions, and observes the resulting rewards. A reward function works in the backend that defines the quality of these executed actions.
+
+### Robotics 
+
+Robots operate in a highly dynamic and ever-changing environment, making it impossible to predict what will happen next. Reinforcement Learning provides a considerable advantage in these scenarios to make the robots robust enough and help acquire complex behaviors adaptively in different scenarios.
+
+It aims to remove the need for time-consuming and tedious checks and replaces them with computer vision systems ensuring higher levels of quality control on the production assembly line.
+
+A wide variety of problems in robotics can be considered as ones of reinforcement learning. Reinforcement learning (RL) enables a robot to autonomously discover an optimal behavior through trial-and-error interactions with its environment. Instead of explicitly detailing the solution to a problem, in reinforcement learning the designer of a control task provides feedback in terms of a scalar objective function that measures the one-step performance of the robot.
 
 ## Task Variants 
 
-This place can be filled with variants of this task if there's any. 
+### Q - Learning
+
+* It is used to learn the policy for reinforcement learning
+* Policy: a rule that the agent should follow to select actions given the current state.
+* Q - learning is to find the optimal policy for making decisions

```

Comment:
isn't Q-learning a value based technique rather than policy based? Also pinging @simoninithomas

Reply:
Q-Learning is valued based but since you find the best value function and the policy is argmax q value you're "indirectly" find an optimal policy function.